demo/full/scripts/controllers/ContentList.tsx

```diff
@@ -276,79 +310,30 @@ function ContentList({
   const [licenseServerUrl, setLicenseServerUrl] = React.useState("");
   const [serverCertificateUrl, setServerCertificateUrl] = React.useState("");
   const [isLowLatencyChecked, setIsLowLatencyChecked] = React.useState(false);
+  const [isAlreadyLoading, setIsAlreadyLoading] = React.useState(false);
 
   /**
-   * Load the given displayed content through the player.
+   * Load the given content through the player.
    * @param {Object|null} content
    */
   const loadContent = React.useCallback(
-    (content: IDisplayedContent) => {
-      const {
-        url,
-        transport,
-        fallbackKeyError,
-        fallbackLicenseRequest,
-        isLowLatency,
-        drmInfos,
-      } = content;
-
-      getKeySystemsOption(drmInfos ?? [], {
-        fallbackKeyError: !!fallbackKeyError,
-        fallbackLicenseRequest: !!fallbackLicenseRequest,
-      }).then(
-        (keySystems) => {
-          loadVideo({
-            url: url ?? "",
-            transport: transport ?? "",
-            textTrackMode: "html",
-            lowLatencyMode: isLowLatency,
-            keySystems,
-          });
-        },
-        () => {
-          /* eslint-disable-next-line no-console */
-          console.error("Could not construct key systems option");
-        },
-      );
+    async (content: IPlayableContent) => {
+      if (!isAlreadyLoading) {
+        setIsAlreadyLoading(true);
+        try {
+          const loadVideoOptions = await getLoadVideoOptions(content);
```

Comment:
To be honest I still feel a race condition is technically possible here.

Perhaps we should have a mechanism (`useRef`) to check that he content is still the last one wanted?

Reply:
To reproduce a race condition, you'll have a first `getLoadVideoOptions` responding very slowly (e.g. server certificate behind a slow CDN) and try to load another content without this issue.

Here we might play the second content, then play the first

Reply:
Oh my bad I didn't see `isAlreadyLoading` is here to fix this. Though I don't think this is the right solution. We want to play the last content, not the initial one.